merlin/models/tf/core/combinators.py

```diff
@@ -523,8 +523,18 @@ def build(self, input_shape):
             if self.automatic_pruning and layer_out_shape == {}:
                 to_prune.append(name)
 
-        for p in to_prune:
-            self.parallel_layers.pop(p)
+        if isinstance(self.parallel_layers, dict):
+            pruned = {}
+            for name, layer in self.parallel_layers.items():
+                if name not in to_prune:
+                    pruned[name] = layer
+            self.parallel_layers = pruned
```

Comment:
Nit: It looks like one of lines 531 or 537 can be removed and the other un-indented to remove duplicate code?

Reply:
Yeah, we need to rewrite this at some point. Right now it’s a bit weird that `self.parallel_layers` could be either of type `List[Layer]` or `Dict[str, Layer]`. Would be simpler to always use the same type, but might be better in a separate PR.